Fix requirements (requests, pytest, setuptools)

speleo3 opened the pull request:
- run requirements
  - add `requests`
  - add `setuptools`
  - remove `pytest` (only needed for tests and already included in test requirements)
- test requirements
  - remove `requests` (moved to run requirements)
  - remove `setuptools`

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * noarch: python recipes are recommended to have a lower bound on the python version. This recommendation will become a requirement in the future.

tsenapathi reviewed (commented):
We are getting the `ModuleNotFoundError: No module named 'pkg_resources'`. Should we keep setuptools?

speleo3 commented:
> We are getting the `ModuleNotFoundError: No module named 'pkg_resources'`. Should we keep setuptools?

Oh I see. I wonder why `conda build -c conda-forge .` passes for me locally :thinking: In that case, we need to add `setuptools` to the run requirements as well, not just test requirements.

speleo3 committed:
Fix requirements (requests, pytest, setuptools)

- run requirements
  - add requests
  - add setuptools
  - remove pytest
- test requirements
  - remove requests
  - remove setuptools
